src/utils/logging.cc

```diff
@@ -80,8 +124,23 @@ StderrLogMessage::StderrLogMessage(const char* file, int line)
     : log_(file, line) {}
 
 StderrLogMessage::~StderrLogMessage() {
-  std::cerr << str() << std::endl;
-  log_ << str();
+  std::string s = str();
+  std::string s2 = "";
+  size_t len;
+  for (;;) {
+    len = s.find("\033[");
+    std::cerr << s.substr(0, len);
+    s2 += s.substr(0, len);
+    if (len == std::string::npos) break;
+    s = s.substr(len);
+    len = 2;
+    while (s[len] && !(s[len] >= 0x40 && s[len] <= 0x7E)) len++;
+    std::cerr.flush();
+    std::cerr << HandleCSI(s.substr(0, ++len));
+    s = s.substr(len);
+  };
+  std::cerr << std::endl;
+  log_ << s2;
```

Comment:
I'm not sure it's a good idea to strip anything (including esc codes from the log).